### PR TITLE
Add documentation for configuring ESLint with Angular applications to the README.

### PR DESCRIPTION
**Pull request by TrevorKarjanis**
- Add documentation for configuring ESLint with Angular applications to the README

Notes
- I had significant issues migrating an Angular application from TSLint that were solved using @angular-eslint.
- Include preliminary documentation for convenience with the expectation of modifications with further review.
- The migration steps install dependencies that may be peer dependencies. See the associated comment.

Tested
- Verified that generating and migrating workspaces and applications works correctly

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,6 +46,72 @@ Extend @ni/eslint-config/typescript in the ESLint configuration. Configure the @
...
+```bash
+> npm i -g @angular/cli @angular-devkit/{core,schematics} @angular-eslint/schematics
+> ng new --collection=@angular-eslint/schematics
+```
+
+#### Migrating a Workspace
+
+1. Install @angular-eslint/schematics and @ni/eslint-config.
+
+    ```npm install --save-dev @angular-eslint/schematics @ni/eslint-config.```
```
Comment (TrevorKarjanis) [inline README.md:67]: These may eventually be peer dependencies. I think we should include this documentation ASAP, so we can discuss these now or leave it for later.

File: README.md [hunk 2]
```diff
@@ -46,6 +46,72 @@ Extend @ni/eslint-config/typescript in the ESLint configuration. Configure the @
...
+
+2. Follow the @angular-eslint [instructions](https://github.com/angular-eslint/angular-eslint#migrating-an-angular-cli-project-from-codelyzer-and-tslint) to run the schematic, listed here for convenience.
+    1. Install the dependencies.
+
+        ```ng add @angular-eslint/schematics```
```
Comment (TrevorKarjanis) [inline README.md:72]: Along with @angular-eslint/schematics, this installs packages that may eventually be peer dependencies. I think we should include this documentation ASAP, so we can discuss these now or leave it for later.

**Review by jattasNI — APPROVED** [review]

**Review by LukasKall — APPROVED** [review]

**Review by mure — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -46,6 +46,49 @@ Extend @ni/eslint-config/typescript in the ESLint configuration. Configure the @
...
+
+ESLint support for Angular is provided by [@angular-eslint](https://github.com/angular-eslint/angular-eslint#readme). It's recommended to use @angular-eslint/schematics to
+configure ESLint for Angular projects especially when migrating from TSLint.
+
+#### Generating a Workspace
+
+Workspaces [generated with @angular-eslint](https://github.com/angular-eslint/angular-eslint#quick-start-with-angular-and-eslint) will generate projects with @angular-eslint.
+
+```bash
+> npm i -g @angular/cli @angular-devkit/{core,schematics} @angular-eslint/schematics
```
Comment (rajsite) [inline README.md:59]: This syntax does not work on windows clis. Can you use a command that is cross-platform?
```
npm i -g @angular/cli @angular-devkit/core @angular-devkit/schematics @angular-eslint/schematics
```

File: README.md [hunk 4]
```diff
@@ -46,6 +46,49 @@ Extend @ni/eslint-config/typescript in the ESLint configuration. Configure the @
...
+    > ng add @angular-eslint/schematics
+    > ng g @angular-eslint/schematics:convert-tslint-to-eslint <PROJECT NAME>
+    ```
+
+3. Extend @ni/eslint-config/typescript in .eslintrc.json for TypeScript and templates.
+    ```json
+    "overrides": [{
+        "files": ["*.ts"],
+            ...
+            "extends": ["@ni/eslint-config/typescript"]
```
Comment (rajsite) [inline README.md:83]: nit: it looks like the indentation went too far, it should be aligned with the "files" key right?

**Review by rajsite — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (TrevorKarjanis) [reply README.md:59]: Thanks for pointing this out. I only ran it on macOS.

File: README.md (hunk 4 above)
Comment (TrevorKarjanis) [reply README.md:83]: Yes, fixed.

**Comment by scotia673** [comment]
FYI: based on some earlier testing, my understanding is that the @angular-eslint tooling is only supported for newer versions of Angular (I think 11+?). Does that jive with your understanding/testing as well? If so, it might be worth calling that out explicitly in the readme.

**Comment by TrevorKarjanis** [comment]
> FYI: based on some earlier testing, my understanding is that the @angular-eslint tooling is only supported for newer versions of Angular (I think 11+?). Does that jive with your understanding/testing as well? If so, it might be worth calling that out explicitly in the readme.

Hey! See the [version support](https://github.com/angular-eslint/angular-eslint#supported-angular-cli-versions) documentation. [1.x.x (documentation)](https://github.com/angular-eslint/angular-eslint/tree/v1.2.0#readme) supports 10.1.0 to <11.2.0 and is still [under development](https://www.npmjs.com/package/@angular-eslint/schematics/v/1.2.1-alpha.12). [2.x.x](https://github.com/angular-eslint/angular-eslint/tree/v2.1.0#readme) supports >=11.2.0. However, [3.x.x](https://www.npmjs.com/package/@angular-eslint/schematics/v/3.0.0-alpha.2) is under development, and Angular may release [12.0](https://www.npmjs.com/package/@angular/cli/v/12.0.0-next.8) next month. @angular-eslint has seen significant development so the latest version will bring advantages. See the Angular [support documentation](https://angular.io/guide/releases#support-policy-and-schedule) for a recommendation on where your product might want to be.

I moved from 10 to 11 last week to make this readme, and as usual it was fine. Obviously, I don't want to include all of this information in this readme, but I do agree it would be helpful to include a note.

Thanks for taking the time!

**Review by rajsite — APPROVED** [review]

**Pull request merged**